### ISSUE TITLE
Custom attributes of images generated by Block\Product\ImageFactory don't render correctly

### ISSUE THREAD

**Issue by chris-pook** [magento/magento2#25219]
###### (i) This ticket is created automatically based on a verified GitHub issue.        
{panel}
***If you are starting to work on this Jira ticket, please notify issue reporter and authors of related community pull requests (if any) that you are going to work on this issue:***
  1.  *Assign [#25219](https://github.com/magento/magento2/issues/25219) issue to yourself on Magento GitHub*
  2.  *Add comment "Internal Magento team is working on this issue right now."*
{panel}
When using the core product image factory it is no longer possible to add your own custom attributes to the markup produced, due to incorrect escaping of the html output.

######Preconditions
1. Install Magento 2.3.3 & 2.3-develop

######Steps to reproduce (*)
1. Within a block or viewModel use the \Magento\Catalog\Block\Product\ImageFactory class to create a Magento\Catalog\Block\Product\Image. Set a custom attribute "data-sku" with the value of the product's sku and render the image in a frontend template:

```
$attributes = <
    'data-sku' => $product->getSku()
>;

$image = $this->imageFactory->create($product, $imageId, $attributes);
```

######Expected result
The image html should render with each of the custom attribute values returned by getCustomAttributes() escaped. In this example `<img data-sku="<sku>" />`.

######Actual result (*)
The image renders with the entire getCustomAttributes() string escaped meaning the html is invalid and the attribute is not available on the html element. In this example `<img &#x20;data-sku&#x3d;&quot;<sku>&quot;  />`.

This stems from an over zealous use of escapeHtmlAttr() here: https://github.com/magento/magento2/commit/f40f24c3ff971fe49480a470868f443b6061d349#diff-eed94ca7983b385a07b8fb5ee2dbc20d

**Comment by engcom-Charlie** [comment]
Hello @chris-pook 
`ImageBuilder` class is deprecated(will be deleted at next release). 
![image](https://user-images.githubusercontent.com/51680850/67691037-97e89400-f9a6-11e9-9cc7-e7a6a48a6909.png)

Please use `ImageFactory` for your actions.
![image](https://user-images.githubusercontent.com/51680850/67690214-2a883380-f9a5-11e9-8323-b8dea39e5ef8.png)

So i have to close this issue.
Thanks for your report!

**Comment by chris-pook** [comment]
Fair enough. Thanks.

**Comment by chris-pook** [comment]
This still seems to be an issue with ImageFactory. The same template vendor/magento/module-catalog/view/frontend/templates/product/image_with_borders.phtml will render the image and this is where the escaping issue is happening.

**Comment by engcom-Charlie** [comment]
@chris-pook could you update the description of issue and add new steps?

**Comment by engcom-Charlie** [comment]
We are closing this issue due to inactivity. If you'd like to update it, please reopen the issue.

**Comment by chris-pook** [comment]
Updated description and description to reflect use of ImageFactory to replicate this issue.

**Comment by engcom-Charlie** [comment]
@chris-pook Thank you for contribution and collaboration!
I can confirm this issue now.
![image_2019_11_14T12_57_14_956Z](https://user-images.githubusercontent.com/51680850/68859402-d6dd5000-06ef-11ea-817b-310b8d2c29a3.png)

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-23137` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @chris-pook. Thank you for your report. 
The issue has been fixed in magento/magento2#26959 by @alexander-aleman in 2.4-develop branch
Related commit(s):
 - [6a8a103ce27d872e5687402a50ba6e594ae82194](https://github.com/magento/magento2/commit/6a8a103ce27d872e5687402a50ba6e594ae82194)
 - [4d3a05d101723432d8610160552b563c25cc3703](https://github.com/magento/magento2/commit/4d3a05d101723432d8610160552b563c25cc3703)
 - [b0ccd4f9fed45147101195d71a7af298d4d8ff4b](https://github.com/magento/magento2/commit/b0ccd4f9fed45147101195d71a7af298d4d8ff4b)
 - [79025f879e94c8b409a7d792588e4fe30ea14f36](https://github.com/magento/magento2/commit/79025f879e94c8b409a7d792588e4fe30ea14f36)
 - [5e84595da92ba3788edac27d8f35bc9a0e0e9d27](https://github.com/magento/magento2/commit/5e84595da92ba3788edac27d8f35bc9a0e0e9d27)
 - [afac7ead70c7c11329c56d3264ee51df655dc995](https://github.com/magento/magento2/commit/afac7ead70c7c11329c56d3264ee51df655dc995)
 - [c9ad76f5aaa100ff676eb52470e2136236b85249](https://github.com/magento/magento2/commit/c9ad76f5aaa100ff676eb52470e2136236b85249)
 - [9fa7ece9756d255cb11aee0a0489d631e36f91c1](https://github.com/magento/magento2/commit/9fa7ece9756d255cb11aee0a0489d631e36f91c1)

The fix will be available with the upcoming 2.4.0 release.
